bitcoin/src/address/mod.rs

```diff
@@ -604,7 +606,7 @@ impl Address {
             Segwit { ref program, hrp: _ } => {
```

Comment:
OT but why don't we store `WitnessProgram` here so that we could just call `ScriptBuf::new_witness_program`?

Reply:
Good idea. I think it's because `Address::Payload` predates `script::WitnessProgram` so it didn't occur to us. But yes, should be done separately to this PR.

Reply:
https://github.com/rust-bitcoin/rust-bitcoin/issues/3012